### PR TITLE
Fix IPv6 detection for EKS

### LINKED ISSUE #36961: Unable to install istio on EKS with IPv6
### Bug Description

Tried my best to install istio on EKS, and everytime `istio-ingressgateway` pod fails to mark as ready, with readiness probe failing with an error log mentioning `Readiness probe failed: Get "http://[IPv6 endpoint of the pod]:15021/healthz/ready": dial tcp [same IPv6 endpoint]:15021: connect: connection refused`. Istioctl keeps waiting for 5 minutes and fails to spin up istio. Is there any IPv6-specific configuration that I have to include to ensure this works?

### Version

```prose
client version: 1.12.2
control plane version: 1.12.2
data plane version: 1.12.2 (1 proxies)

Client Version: v1.23.2
Server Version: v1.21.5-eks-bc4871b
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1

Didnt use helm
```

### Additional Information

Target cluster context: devops@CLUSTER_NAME.ap-south-1.eksctl.io

Running with the following config:

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-system,kube-public,kube-node-lease,local-path-storage }
end-time: 2022-01-24 16:45:47.136144463 +0530 IST

Cluster endpoint: https://HIDDEN.gr7.ap-south-1.eks.amazonaws.com
CLI version:
version.BuildInfo{Version:"1.12.2", GitRevision:"af0d66fd0aa363e9a7b0164f3a94ba36252fe60f", GolangVersion:"go1.17.5", BuildStatus:"Clean", GitTag:"1.12.2"}

The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Info:      version.BuildInfo{Version:"1.12.2", GitRevision:"af0d66fd0aa363e9a7b0164f3a94ba36252fe60f-dirty", GolangVersion:"", BuildStatus:"Modified", GitTag:"1.12.2"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.12.2}

Fetching proxy logs for the following containers:

istio-system/istio-ingressgateway/istio-ingressgateway-66cc5dcb7d-tgfwf/istio-proxy
istio-system/istiod/istiod-577b69955f-nx62b/discovery

Fetching Istio control plane information from cluster.

Running istio analyze on all namespaces and report as below:
Analysis Report:
Info [IST0102] (Namespace cert-manager) The namespace is not enabled for Istio injection. Run 'kubectl label namespace cert-manager istio-injection=enabled' to enable it, or 'kubectl label namespace cert-manager istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace default) The namespace is not enabled for Istio injection. Run 'kubectl label namespace default istio-injection=enabled' to enable it, or 'kubectl label namespace default istio-injection=disabled' to explicitly mark it as not needing injection.
Creating an archive at /home/baalajimaestro/bug-report.tar.gz.
Cleaning up temporary files in /tmp/bug-report.
Done.

### PR DESCRIPTION
**Pull request by mKeRix**
**Please provide a description of this PR:**

EKS clusters on IPv6 map a host-local NAT interface to each pod for IPv4 egress. For that the pod will be assigned an IP from the link-local reserved CIDR range (more specifcally, from `169.254.172.0/22`) that is unique only to the host. These IPs are not advertised to Kubernetes in any way, they are just available within the pod as the `v4if0` network interface. For proper IPv6 detection we need to exclude these IPs from the discovery mechanisms, otherwise Istio thinks that it is in an IPv4 cluster when it sees this network interface.

More details on the EKS IPv6 networking can be found in the [official blog post](https://aws.amazon.com/de/blogs/containers/amazon-eks-launches-ipv6-support/).

Unfortunately, I did not find existing tests for this code - if there are any that I have overlooked do let me know and I will update the pull request accordingly.

Fixes #36961.

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Review by bianpengyuan — COMMENTED** [review]
LGTM, will let others to take a look. Do we also want to exclude link local multicast addresses?

**Comment by bianpengyuan** [comment]
Also I think this worth a release notes.

**Comment by mKeRix** [comment]
@bianpengyuan Added the release note now. I think excluding link-local multicast makes sense, I initially didn't include in this PR to keep it as small as possible to get the bug fixed. However, I'm happy to add it either directly to this PR or to open an additional followup PR - just let me know what you'd prefer.

**Comment by bianpengyuan** [comment]
> I'm happy to add it either directly to this PR

@mKeRix I think it is fine to include that directly here

**Comment by mKeRix** [comment]
Done!

**Comment by howardjohn** [comment]
This seems reasonable but I don't have enough expertise to approve. @ramaraochavali @lambdai @costinm @hzxuzhonghu any concerns?

**Comment by jacob-delgado** [comment]
We had to do this for implementing dual-stack which we will have more information shortly. LGTM, but I'll wait to approve after the networking maintainers approve.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
